services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -845,6 +846,56 @@ public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_P
         }
     }
 
+     /**
+     * Generate action tokens
+     *
+     * An token to perform a set of required actions. The clientId parameters are
+     * optional.
+     *
+     * @param clientId Client id
+     * @param lifespan Number of seconds after which the generated token expires
+     * @param actions  required actions the user needs to complete
+     * @return ActionToken
+     */
+    @Path("generate-actions-token")
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.CLIENT_ID_PARAM) String clientId,
```

Comment:
Instead of replicating code from `org.keycloak.services.resources.admin.UserResource#executeActionsEmail` we could probably reuse the same code that performs the validation and token creation.

I would also change the method name to `createActionToken` or something.

Reply:
Outsourcing of validation and token generation to `createActionToken` function.